### PR TITLE
E2E BERT Model

### PR DESCRIPTION
**Pull request by karthik-77**
Signed-off-by: karthik-77 <karthiks@ideas2it.com>

## What changes are proposed in this pull request?

End to End deep learning workflow using BERT model has been added

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow TorchServe Deployment Plugin users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?
Components
- [ ] `area/deploy`: Main deployment plugin logic
- [ ] `area/build`: Build and test infrastructure for MLflow TorchServe Deployment Plugin
- [ ] `area/docs`: MLflow TorchServe Deployment Plugin documentation pages
- [X] `area/examples`: Example code

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/E2EBert/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+
+For example:
+```
+mlflow run . -P max_epochs=5 num-samples=50000
+```
+
+Or to run the training script directly with custom parameters:
+```
+python mnist_autolog_example1.py \
```
Comment (HamidShojanazeri) [inline examples/E2EBert/README.md:67]: @karthik-77 Thanks for the PR, here the doc is referring to Mnist_autolog example, which needs to be changed to news_classifier.py. Also, adding the command to run with a gpu might be helpful.

Comment (karthik-77) [reply examples/E2EBert/README.md]: Have incorporated the suggested change in readme file

File: examples/E2EBert/README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+## Package Requirement
+
+Install the required packages using the following command
+
+`pip install -r requirements.txt`
+ 
+ 
+ ### Running the code
+To run the example via MLflow, navigate to the `mlflow-torchserve/examples/BertNewsClassification_E2E` directory and run the command
```
Comment (chauhang) [inline examples/E2EBert/README.md:20]: @karthik-77 Please fix the folder name, it does not match the actual name of the folder

**Comment by shrinath-suresh** [comment]
@karthik-77 Can you please add the example in unit test case - with epochs as 1 and num samples as 100?

File: examples/E2EBert/README.md (hunk 2 above)
Comment (karthik-77) [reply examples/E2EBert/README.md]: have changed the folder name in readme file

**Review by shrinath-suresh — CHANGES_REQUESTED** [review]

File: examples/E2EBert/conda.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+channels:
+- defaults
+- conda-forge
+- pytorch
+dependencies:
+- python=3.8.2
+- pytorch>=1.6.0
+- torchvision=0.7.0
```
[the remark this answers is not in the document]
Comment (karthik-77) [reply examples/E2EBert/conda.yaml]: have unpinned the versions

File: examples/E2EBert/conda.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+- pip
+- pip:
+  - mlflow
+  - gorilla
+  - sklearn
+  - cloudpickle==1.6.0
+  - boto3
+  - transformers
+  - torchtext==0.7.0
```
[the remark this answers is not in the document]
Comment (karthik-77) [reply examples/E2EBert/conda.yaml]: have unpinned the `torchtext` version

File: examples/E2EBert/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+torch==1.6.0
+pytorch_lightning==1.0.4
```
[the remark this answers is not in the document]
Comment (karthik-77) [reply examples/E2EBert/requirements.txt]: have incorporated the suggested changes

File: examples/E2EBert/news_classifier_handler.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+    def postprocess(self, inference_output):
+        """
+        Does postprocess after inference to be returned to user
+
+        :param inference_output: Output of inference
+
+        :return: output - Output after post processing
+        """
+        if self.class_mapping_file:
```
[the remark this answers is not in the document]
Comment (karthik-77) [reply examples/E2EBert/news_classifier_handler.py]: have modified the handler code accordingly

File: examples/E2EBert/README.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    --max_epochs 5 \
+    --num_samples 50000 \
+```
+
+
+## Installing Deployment Plugin
+
+Run the following commands to install deployment plugin
+
+`python setup.py build`
```
[the remark this answers is not in the document]
Comment (karthik-77) [reply examples/E2EBert/README.md]: Have removed the installation steps

**Review by chauhang — APPROVED** [review]

**Review by shrinath-suresh — APPROVED** [review]

**Review by HamidShojanazeri — APPROVED** [review]

**Comment by shrinath-suresh** [comment]
@HamidShojanazeri Can you please sign off the commits ?

**Pull request merged**